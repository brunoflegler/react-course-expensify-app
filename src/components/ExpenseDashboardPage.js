import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpenseSummaryPage from './ExpenseSummaryPage';


export const ExpenseDashboardPage = () => (
    <div>
        <ExpenseSummaryPage />
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);

export default ExpenseDashboardPage;