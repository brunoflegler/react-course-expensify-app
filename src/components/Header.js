import React from 'react'
import { NavLink } from 'react-router-dom';

const Header = () => (
    <header className="header">
        <h1>Expensify</h1>
        <NavLink exact={true} to="/" activeClassName="is-active" > Go home </NavLink>&nbsp;&nbsp;&nbsp;
        <NavLink to="/create" activeClassName="is-active" > Create </NavLink>&nbsp;&nbsp;&nbsp;
        <NavLink to="/edit" activeClassName="is-active" > Edit </NavLink>&nbsp;&nbsp;&nbsp;
        <NavLink to="/help" activeClassName="is-active" > Help </NavLink>&nbsp;&nbsp;&nbsp;
    </header>
)



export default Header;