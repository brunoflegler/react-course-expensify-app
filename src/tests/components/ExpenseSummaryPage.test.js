import React from 'react';
import { shallow } from 'enzyme'
import { ExpenseSummaryPage } from '../../components/ExpenseSummaryPage';



test('should correctly render ExpenseSummaryPage with 1 expense', () => {
    const wrapper = shallow(<ExpenseSummaryPage expensesCount={1} expensesTotal={235} />  );
    expect(wrapper).toMatchSnapshot();
});



test('should correctly render ExpenseSummary with 1 expense', () =>{
    const wrapper = shallow(<ExpenseSummaryPage expensesCount={23} expensesTotal={23512340987} />  );
    expect(wrapper).toMatchSnapshot();
});