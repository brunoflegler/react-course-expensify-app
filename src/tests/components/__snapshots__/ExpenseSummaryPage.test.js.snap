// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpenseSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing  
    23
     
    expenses
      totalling R$ 
    235,123,409.87
     
  </h1>
</div>
`;

exports[`should correctly render ExpenseSummaryPage with 1 expense 1`] = `
<div>
  <h1>
    Viewing  
    1
     
    expense
      totalling R$ 
    2.35
     
  </h1>
</div>
`;
